options/cf_options.cc

```diff
@@ -297,10 +375,69 @@ std::unordered_map<std::string, OptionTypeInfo>
          {offset_of(&ColumnFamilyOptions::memtable_factory),
           OptionType::kMemTableRepFactory, OptionVerificationType::kByName,
           OptionTypeFlags::kNone, 0}},
+        {"memtable",
+         {offset_of(&ColumnFamilyOptions::memtable_factory),
+          OptionType::kMemTableRepFactory, OptionVerificationType::kAlias,
+          OptionTypeFlags::kNone, 0,
+          [](const ConfigOptions& /*opts*/, const std::string& /*name*/,
+             const std::string& value, char* addr) {
+            std::unique_ptr<MemTableRepFactory> rep;
+            Status s = GetMemTableRepFactoryFromString(value, &rep);
+            if (s.ok()) {
+              auto mtf =
+                  reinterpret_cast<std::shared_ptr<MemTableRepFactory>*>(addr);
+              mtf->reset(rep.release());
+            }
+            return s;
+          }}},
         {"table_factory",
          {offset_of(&ColumnFamilyOptions::table_factory),
           OptionType::kTableFactory, OptionVerificationType::kByName,
-          OptionTypeFlags::kNone, 0}},
+          OptionTypeFlags::kCompareLoose, 0}},
+        {"block_based_table_factory",
+         {offset_of(&ColumnFamilyOptions::table_factory),
+          OptionType::kTableFactory, OptionVerificationType::kAlias,
+          OptionTypeFlags::kCompareLoose, 0,
+          [](const ConfigOptions& /*opts*/, const std::string& /*name*/,
+             const std::string& value, char* addr) {
+            // Nested options
+            auto tf = reinterpret_cast<std::shared_ptr<TableFactory>*>(addr);
+            BlockBasedTableOptions table_opts, base_opts;
+            BlockBasedTableFactory* bbtf =
+                static_cast_with_check<BlockBasedTableFactory, TableFactory>(
+                    tf->get());
+            if (bbtf != nullptr) {
+              base_opts = bbtf->table_options();
+            }
+            Status s = GetBlockBasedTableOptionsFromString(base_opts, value,
```

Comment:
If we based on the original logic if !GetBlockBasedTableOptionsFromString().ok(), it should returns: 
```
return Status::InvalidArgument(
            "unable to parse the specified CF option " + name);
```
But here, s is directly returned. The behavior will be different from the current code. Other cases such as  plain_table_factory, memtable also has the similar mismatches.

Reply:
In the original code, you would lose whatever error you got from the lower-level call.  For example, if you passed in a bad block-based table option, you would just get "Unable to parse block_based_table_factory" and have no idea or means to figure out why.  This code preserves the same error semantics -- returns success or failure under the same scenarios -- but gives the true underlying reason for the error rather than a generic, ambiguous wrapper.

The semantics are the same as the existing code but we return a more informative error.  Isn't that a good thing?

Reply:
> In the original code, you would lose whatever error you got from the lower-level call. For example, if you passed in a bad block-based table option, you would just get "Unable to parse block_based_table_factory" and have no idea or means to figure out why. This code preserves the same error semantics -- returns success or failure under the same scenarios -- but gives the true underlying reason for the error rather than a generic, ambiguous wrapper.
> 
> The semantics are the same as the existing code but we return a more informative error. Isn't that a good thing?

I see. Now I got your argument. It would be good to return the lower-level call status.